### ISSUE TITLE
Weakly referenced object issue

### ISSUE THREAD

**Issue by banesullivan** [vtkiorg/vtki#158]
https://github.com/vtkiorg/vtki/blob/b12872c26ba78d57419dae974de0e104ab3636c7/vtki/common.py#L851

From time to time when using objects nested in a `vtki.MultiBlock`, I get a `ReferenceError` on the `self.data` attribute of the `_ScalarsDict` class.

Is this proxy cast necessary? What's the point? Could we make it not weakly referenced (`<class 'weakproxy'>`?

```py
---------------------------------------------------------------------------
ReferenceError                            Traceback (most recent call last)
<ipython-input-5-1ae88bf5bf88> in <module>()
      1 # Add values to the meshes so colors are consistent with other data
      2 # try:
----> 3 proj['temp_175c'].point_arrays['Temperature'] = np.full(proj['temp_175c'].n_points, 175.)
      4 proj['temp_225c'].point_arrays['Temperature'] = np.full(proj['temp_225c'].n_points, 225.)
      5 #     tc = vtki.MultiBlock()

~/Documents/OpenGeoVis/Software/vtki/vtki/common.py in __setitem__(self, key, val)
    883         """ overridden to assure data is contigious """
    884         if self.callback_enabled:
--> 885             self.adder(val, key, deep=False)
    886         dict.__setitem__(self, key, val)
    887         self.modifier()

~/Documents/OpenGeoVis/Software/vtki/vtki/common.py in adder(self, scalars, name, set_active, deep)
    920 
    921     def adder(self, scalars, name, set_active=False, deep=True):
--> 922         self.data._add_point_scalar(scalars, name, set_active=False, deep=deep)
    923 
    924 

ReferenceError: weakly-referenced object no longer exists
```

**Comment by banesullivan** [comment]
Note that if I change my code to:
```py
temp_225c = proj['temp_225c']
temp_175c.point_arrays['Temperature'] = np.full(proj['temp_175c'].n_points, 175.)
```
it works.....

**Comment by akaszynski** [comment]
It works, but the objects may not be collected if there are internal references.  We might want to introduce some tests to make sure they collect.

**Comment by akaszynski** [comment]
Might consider depreciating ``point_arrays`` and ``cell_arrays`` for ``__getitem__`` and ``__setitem__``.

**Comment by banesullivan** [comment]
> Might consider depreciating `point_arrays` and `cell_arrays` for `__getitem__` and `__setitem__`

Per #161, I don't think we should do this - also `__getitem__` and `__setitem__` simply call back to `point_arrays` and `cell_arrays` depending on the array size. The difference between point/cell data is heavily stressed in VTK and its a really import concept for most of the data I deal with where data resolution needs to be conveyed. 

I think users should be aware of these differences between points/cells and advanced users should stick with using the `point_arrays` and `cell_arrays` properties. I'd like for `__getitem__` and `__setitem__` to be simply for convenience to help new users get up and running.

**Comment by banesullivan** [comment]
This reference error is repeatable anytime when trying to assign data to an object within a `vtki.MultiBlock` dataset that has already been accessed.

```py
from vtki import examples
import vtki
import numpy as np

blocks = vtki.MultiBlock([
                          examples.load_uniform(),
                          examples.load_structured(),
                         ])

# Access data in the multiblock dataset once
arr = np.random.rand(blocks[0].n_points)
# Try to edit the data dictionaries
blocks[0].point_arrays['foo'] = arr
```
```py
---------------------------------------------------------------------------
ReferenceError                            Traceback (most recent call last)
<ipython-input-10-1bd4391ab7a2> in <module>()
      9 
     10 arr = np.random.rand(blocks[0].n_points)
---> 11 blocks[0].point_arrays['foo'] = arr

~/Documents/OpenGeoVis/Software/vtki/vtki/common.py in __setitem__(self, key, val)
    914         """ overridden to assure data is contigious """
    915         if self.callback_enabled:
--> 916             self.adder(val, key, deep=False)
    917         dict.__setitem__(self, key, val)
    918         self.modifier()

~/Documents/OpenGeoVis/Software/vtki/vtki/common.py in adder(self, scalars, name, set_active, deep)
    951 
    952     def adder(self, scalars, name, set_active=False, deep=True):
--> 953         self.data._add_point_scalar(scalars, name, set_active=False, deep=deep)
    954 
    955 

ReferenceError: weakly-referenced object no longer exists
```

**Comment by akaszynski** [comment]
I agree about not removing ``point_arrays`` and ``cell_arrays``, and I still think that it would be great to add a getter and setter to the ``common`` object for convenience.  It's something I'll probably end up using as it's a bit of work to write ``point_arrays`` for much of my closed-source dependent code.

**Comment by akaszynski** [comment]
Fixed the reference error in 83fa649.

**Comment by banesullivan** [comment]
Thanks, Alex!
